casper/src/test/scala/coop/rchain/casper/helper/RhoSpec.scala

```diff
@@ -101,6 +103,37 @@ object RhoSpec {
         } yield result
       }
     }
+
+  private def setupRuntime[F[_]: Sync](
+      runtime: Runtime[F],
+      otherLibs: Seq[DeployData]
+  ): F[Runtime[F]] =
+    for {
+      _ <- evalDeploy(rhoSpecDeploy, runtime)
+      _ <- otherLibs.toList.traverse(evalDeploy(_, runtime))
+      // reset the deployParams.userId before executing the test
+      // otherwise it'd execute as the deployer of last deployed contract
+      _ <- runtime.deployParametersRef.update(_.copy(userId = Par()))
+    } yield runtime
+
+  private def evalDeploy[F[_]: Sync](
+      deploy: DeployData,
+      runtime: Runtime[F]
+  ): F[Unit] = {
+    val rand: Blake2b512Random = Blake2b512Random(
+      DeployData.toByteArray(ProtoUtil.stripDeployData(deploy))
+    )
+    eval(deploy.term, runtime, NormalizerEnv(deploy))(implicitly, rand)
+  }
+
+  private val rhoSpecDeploy: DeployData =
+    DeployData(
+      deployer = ProtoUtil.stringToByteString(
```

Comment:
Does ConstructDeploy.defaultPub work here?

Reply:
Will try

Reply:
Actually, it doesn't make sense - the rho spec contract should be deployed just like any other contract, and not by the default deployer used in tests. Having this say 'defaultSec' would suggest a connection where there isn't.